### PR TITLE
xkbcommon: add components

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **xkbcommon/***

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by uilianries** [comment]
@ericLemanissier It would be nice if you exercise the components in test_package/CMakeLists.txt, like adding `find_package(xkbcommon REQUIRED); target_link_libraries(app xkbcommon::xkbcommon)`

**Comment by ericLemanissier** [comment]
the pkg-config test fails because CCI is running conan 1.27 which does not have https://github.com/conan-io/conan/pull/7413
It will also fail with 1.28.0 because of https://github.com/conan-io/conan/issues/7474. it should pass on 1.28.1

**Comment by ericLemanissier** [comment]
I reverted the components pkg-config test, so that it passes on conan 1.27, and can be merged without waiting conan 1.28.1 to be deployed on CCI

File: recipes/xkbcommon/all/conanfile.py [hunk 1]
```diff
@@ -73,5 +73,12 @@ def package(self):
...
     def package_info(self):
-        self.cpp_info.libs = tools.collect_libs(self)
+        self.cpp_info.components['libxkbcommon'].libs = ['xkbcommon']
+        self.cpp_info.components['libxkbcommon'].name = 'xkbcommon'
+        self.cpp_info.components['libxkbcommon'].requires = ['xorg::xorg']
+        if self.options.with_x11:
+            self.cpp_info.components['libxkbcommon-x11'].libs = ['xkbcommon-x11']
+            self.cpp_info.components['libxkbcommon-x11'].name = 'xkbcommon-x11'
+            self.cpp_info.components['libxkbcommon-x11'].requires = ['libxkbcommon']
+            self.cpp_info.components['libxkbcommon-x11'].requires.append('xorg::xorg')
```
Comment (jgsogo) [inline recipes/xkbcommon/all/conanfile.py:83]:
```suggestion
```

This line shouldn't be needed, it is a transitive dependency via component `libxkbcommon`

File: recipes/xkbcommon/all/test_package/conanfile.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 class TestPackageConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake"
+    generators = "cmake", "cmake_find_package_multi", "pkg_config"
```
Comment (jgsogo) [inline recipes/xkbcommon/all/test_package/conanfile.py:7]: `pkg_config`, is it used?

File: recipes/xkbcommon/all/conanfile.py (hunk 1 above)
Comment (ericLemanissier) [reply recipes/xkbcommon/all/conanfile.py:83]: `libxkbcommon` and `libxkbcommon-x11` do not use the same xorg components. The end goal is to do ` self.cpp_info.components['libxkbcommon'].requires.append('xorg::xkeyboard-config')` and ` self.cpp_info.components['libxkbcommon-x11'].requires.extend(['xorg::xcb', xorg::xcb-xkb'])` when https://github.com/conan-io/conan-center-index/pull/2362 is merged

Comment (jgsogo) [reply recipes/xkbcommon/all/conanfile.py:83]: Merged! 😉 

Maybe we can update this PR accordingly, thanks!

**Pull request merged**